### ISSUE TITLE
Installing elm on Ubuntu

### ISSUE THREAD

**Issue by katbow** [dwyl/learn-elm#30]
This seems like quite the process. I'm wondering if:

1. I am missing something and there's a smoother way to install

2. If the process I'm following is correct, then what is the best way to include the instructions for installing on Ubuntu in the repo

I am *mostly* following [this](https://alphydan.svbtle.com/elm-lang-on-ubuntu-14-04) but with some minor tweaks. It seems rather long to include all of that directly in the readme, so perhaps I can make an updated version in another repo and it can be linked to? Ideally there is a quicker install process but doesn't seem so.

P.S. I tried `npm install -g elm` and that doesn't work :cry:

**Comment by iteles** [comment]
@katbow I've never had to try this so I can't help with the long-windedness itself but I would suggest putting your updates into *this* repo, just in a separate md file that can be linked to from the main readme. 

Seems like a pain!

**Comment by SeanGallen** [comment]
@katbow I agree with @iteles.  I found some info but warning it is long winded. 

I did a little research on Elm's homepage, http://elm-lang.org/.  It referred me to the source at https://github.com/elm-lang/elm-platform.  Here I found the following statement. 
__________________________________________________________________________________________________
First, if you are on Mac or Windows just looking to use Elm, definitely use the installers linked here. Do not put yourself through this process if you do not have to!

Second, if you are on some linux OS and just want to use Elm, use the npm installer.

Finally, if you have made it this far, you are in some unique position where the other options do not cover you for some reason. You are about to actually build from source. I recommend you read this whole section before you start running anything.

________________________________________________________________________________________________

Before I jumped into any of that I would check everything else is working ok. (i.e. npm ). I hope it helps.

**Comment by katbow** [comment]
Got it installed now - it was an npm problem :unamused: 

I'll still write up what I found in another md file that we can link to for installing on Ubuntu :+1:

**Comment by lucymonie** [comment]
Hey @katbow I would love your advice on this as I have been tearing my hair out with it this afternoon. 

I'm also on Ubuntu. I thought I had Elm installed (I used ```npm install -g elm```) but some syntax in elm-repl wasn't working (function syntax), so I figure I must have installed an old version. I followed the instructions on https://alphydan.svbtle.com/elm-lang-on-ubuntu-14-04 but it choked horribly so I've just uninstalled it all again. Any ideas?

**Comment by lucymonie** [comment]
After all that I tried ```npm install -g elm``` again and then ```npm install``` and this seems to have fixed it :yellow_heart:. An hour of my life I won't get back but at least it's not some horrible Ubuntu problem!

**Comment by katbow** [comment]
@lucymonie Glad you got it solved! My specific issue was that I had recently wiped my computer and reinstalled everything, but I realised I didn't change my npm permissions. I watched [this video](https://docs.npmjs.com/getting-started/fixing-npm-permissions) and fixed them from there.

**Comment by jruts** [comment]
@katbow and @lucymonie , if you install node on your ubuntu machine with [nvm](https://github.com/creationix/nvm) you should not run into any npm permission issues.

So I am curious why you had to change your npm permissions?

**Comment by PssThemes** [comment]
from @tino415 , [on this issue](https://github.com/elm-lang/elm-platform/issues/218)
"""
I had same problem, but solved it by running 

`sudo npm install -g elm --unsafe-perm=true --allow-root` 
```
ubuntu 16.04
node -v v7.10.0
npm -v 5.3.0
```
"""

it worked for me, maybe helps someone :smile:
